substrate/frame/nomination-pools/src/lib.rs

```diff
@@ -1078,6 +1088,17 @@ impl<T: Config> BondedPool<T> {
 		self.is_root(who)
 	}
 
+	fn can_claim_commission(&self, who: &T::AccountId) -> bool {
+		if let Some(permission) = self.commission.claim_permission.as_ref() {
+			match permission {
+				CommissionClaimPermission::Permissionless => true,
+				CommissionClaimPermission::Account(account) => account == who || self.is_root(who),
+			}
+		} else {
+			self.is_root(who)
+		}
+	}
```

Comment:
nit or probably just preference but I would check `self.is_root` first and break early if that's the case.